# E-commerce Sales Analysis

## Objective
Create a complete data analysis notebook using **Python and Pandas** for the provided e-commerce sales dataset.

### Analysis covered
- Loading the CSV dataset into a Pandas DataFrame
- Inspecting the dataset
- Selecting columns and records
- Filtering orders
- Sorting sales records
- Using `groupby()` and aggregation functions
- Analyzing sales by category, city, product, region, and payment method
- Calculating total sales, average sales, highest and lowest sales
- Calculating total quantity sold and number of orders
- Identifying top-performing products and categories
- Writing meaningful observations based on the findings


## 1. Import Pandas

In [1]:
import pandas as pd
import numpy as np

print("Pandas imported successfully.")

Pandas imported successfully.


## 2. Load the Dataset

In [2]:
# Upload Day8_Ecommerce_Sales_Dataset.csv to Google Colab before running this cell.
df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

print("Dataset loaded successfully.")
print("Shape:", df.shape)

Dataset loaded successfully.
Shape: (100, 13)


## 3. Inspect the Dataset

In [3]:
print("First 5 rows:")
print(df.head())

print("\nLast 5 rows:")
print(df.tail())

print("\nColumns:")
print(df.columns.tolist())

print("\nData types:")
print(df.dtypes)

First 5 rows:
 Order_ID Order_Date     Customer      City Region    Category             Product  Quantity  Unit_Price  Discount_Percent  Total_Sales  Rating   Payment_Method
     1001 2026-02-27    Kabir Ali Hyderabad  South Electronics Wireless Headphones         3        1499                 0      4497.00     3.8 Cash on Delivery
     1002 2026-01-24 Aditya Verma     Jammu  North      Sports       Running Shoes         1        2799                 0      2799.00     4.1 Cash on Delivery
     1003 2026-04-18   Sara Ahmed   Chennai  South Electronics          Power Bank         2        1199                15      2038.30     4.6 Cash on Delivery
     1004 2026-03-13    Kabir Ali   Lucknow  North Electronics Mechanical Keyboard         4        2499                 5      9496.20     4.1       Debit Card
     1005 2026-03-30  Priya Menon     Jammu  North Electronics         Smart Watch         1        3299                 5      3134.05     4.2              UPI

Last 5 rows:
 Order

## 4. Basic Dataset Information

In [4]:
print("Number of rows:", df.shape[0])
print("Number of columns:", df.shape[1])

print("\nDataset information:")
df.info()

Number of rows: 100
Number of columns: 13


## 5. Check for Missing Values

In [5]:
missing_values = df.isnull().sum()
print(missing_values)

Order_ID            0
Order_Date          0
Customer            0
City                0
Region              0
Category            0
Product             0
Quantity            0
Unit_Price          0
Discount_Percent    0
Total_Sales         0
Rating              0
Payment_Method      0


## 6. Select Specific Columns

In [6]:
# Selecting individual columns
print("Product and Total Sales:")
print(df[["Product", "Total_Sales"]].head(10))

# Selecting multiple useful order details
print("\nSelected order details:")
print(df[["Order_ID", "City", "Category", "Quantity", "Total_Sales"]].head(10))

Product and Total Sales:
            Product  Total_Sales
Wireless Headphones      4497.00
      Running Shoes      2799.00
         Power Bank      2038.30
Mechanical Keyboard      9496.20
        Smart Watch      3134.05
      The Alchemist       449.10
       Water Bottle      1398.00
             Hoodie      6076.20
    Electric Kettle      4497.00
             Jacket      8996.40

Selected order details:
 Order_ID      City       Category  Quantity  Total_Sales
     1001 Hyderabad    Electronics         3      4497.00
     1002     Jammu         Sports         1      2799.00
     1003   Chennai    Electronics         2      2038.30
     1004   Lucknow    Electronics         4      9496.20
     1005     Jammu    Electronics         1      3134.05
     1006     Kochi          Books         1       449.10
     1007    Jaipur Home & Kitchen         2      1398.00
     1008     Jammu       Clothing         4      6076.20
     1009    Mumbai Home & Kitchen         3      4497.00
     10

## 7. Filter Records

In [7]:
# Orders with sales greater than 10,000
high_sales = df[df["Total_Sales"] > 10000]
print("Orders with Total_Sales > 10000:")
print(high_sales[["Order_ID", "Customer", "Product", "Total_Sales"]])

# Electronics orders
electronics = df[df["Category"] == "Electronics"]
print("\nNumber of Electronics orders:", len(electronics))

# UPI orders
upi = df[df["Payment_Method"] == "UPI"]
print("Number of UPI orders:", len(upi))

Orders with Total_Sales > 10000:
 Order_ID     Customer       Product  Total_Sales
     1013  Maryam Khan        Jacket     10620.75
     1048   Sara Ahmed  Coffee Maker     10497.00
     1049 Ishita Gupta Running Shoes     13995.00
     1053 Ananya Singh Running Shoes     13295.25
     1085   Sana Malik  Coffee Maker     12596.40
     1090 Aditya Verma   Cricket Bat     12495.00

Number of Electronics orders: 19
Number of UPI orders: 26


## 8. Apply Multiple Filters

In [8]:
# Sports orders with sales greater than 5,000
filtered_orders = df[
    (df["Category"] == "Sports") &
    (df["Total_Sales"] > 5000)
]

print(filtered_orders[
    ["Order_ID", "City", "Product", "Quantity", "Total_Sales"]
])

 Order_ID      City       Product  Quantity  Total_Sales
     1014    Jaipur Running Shoes         3      8397.00
     1017 Hyderabad   Cricket Bat         3      7497.00
     1038      Pune  Dumbbell Set         5      9495.25
     1049    Mumbai Running Shoes         5     13995.00
     1053 Hyderabad Running Shoes         5     13295.25
     1059   Chennai   Cricket Bat         4      8496.60
     1060      Pune  Dumbbell Set         3      5397.30
     1090   Lucknow   Cricket Bat         5     12495.00
     1093 Hyderabad   Cricket Bat         4      8496.60
     1096     Kochi  Dumbbell Set         5      9495.25


## 9. Sort Records by Sales

In [9]:
# Highest-value orders
top_orders = df.sort_values("Total_Sales", ascending=False).head(10)

print("Top 10 orders by Total Sales:")
print(top_orders[
    ["Order_ID", "Customer", "City", "Product", "Quantity", "Total_Sales"]
])

# Lowest-value orders
print("\nLowest 5 orders by Total Sales:")
print(df.sort_values("Total_Sales").head(5)[
    ["Order_ID", "Customer", "Product", "Total_Sales"]
])

Top 10 orders by Total Sales:
 Order_ID     Customer       City             Product  Quantity  Total_Sales
     1049 Ishita Gupta     Mumbai       Running Shoes         5     13995.00
     1053 Ananya Singh  Hyderabad       Running Shoes         5     13295.25
     1085   Sana Malik  Hyderabad        Coffee Maker         4     12596.40
     1090 Aditya Verma    Lucknow         Cricket Bat         5     12495.00
     1013  Maryam Khan    Lucknow              Jacket         5     10620.75
     1048   Sara Ahmed    Lucknow        Coffee Maker         3     10497.00
     1070 Aditya Verma      Delhi              Jacket         4      9996.00
     1080 Aditya Verma      Delhi              Jacket         4      9996.00
     1052  Rohan Mehta Chandigarh Mechanical Keyboard         4      9496.20
     1004    Kabir Ali    Lucknow Mechanical Keyboard         4      9496.20

Lowest 5 orders by Total Sales:
 Order_ID   Customer            Product  Total_Sales
     1006  Manya Rao      The Alchemi

## 10. Overall Sales Statistics

In [10]:
total_sales = df["Total_Sales"].sum()
average_sales = df["Total_Sales"].mean()
highest_sales = df["Total_Sales"].max()
lowest_sales = df["Total_Sales"].min()
total_quantity = df["Quantity"].sum()
number_of_orders = df["Order_ID"].nunique()

print(f"Total Sales: ₹{total_sales:,.2f}")
print(f"Average Sales per Order: ₹{average_sales:,.2f}")
print(f"Highest Sale: ₹{highest_sales:,.2f}")
print(f"Lowest Sale: ₹{lowest_sales:,.2f}")
print(f"Total Quantity Sold: {total_quantity}")
print(f"Number of Orders: {number_of_orders}")

Total Sales: ₹447,262.95
Average Sales per Order: ₹4,472.63
Highest Sale: ₹13,995.00
Lowest Sale: ₹449.10
Total Quantity Sold: 294
Number of Orders: 100


## 11. Sales Analysis by Category

In [11]:
category_analysis = df.groupby("Category").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Highest_Sale=("Total_Sales", "max"),
    Lowest_Sale=("Total_Sales", "min"),
    Quantity_Sold=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values("Total_Sales", ascending=False)

print(category_analysis)

                Total_Sales  Average_Sales  Highest_Sale  Lowest_Sale  Quantity_Sold  Number_of_Orders
Category                                                                                              
Sports            136026.45        5441.06      13995.00       759.05             79                25
Home & Kitchen    113329.90        5396.66      12596.40      1188.30             59                21
Electronics        92086.00        4846.63       9496.20      1708.10             57                19
Clothing           69028.20        4930.59      10620.75       679.15             39                14
Books              36792.40        1752.02       4495.00       449.10             60                21


## 12. Sales Analysis by City

In [12]:
city_analysis = df.groupby("City").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Quantity_Sold=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values("Total_Sales", ascending=False)

print(city_analysis)

             Total_Sales  Average_Sales  Quantity_Sold  Number_of_Orders
City                                                                    
Hyderabad       59076.45        8439.49             25                 7
Chennai         47493.20        4749.32             29                10
Lucknow         46194.65        7699.11             22                 6
Pune            41908.55        4190.86             34                10
Mumbai          41457.80        5182.23             23                 8
Kochi           37257.50        4139.72             29                 9
Chandigarh      34539.65        4317.46             22                 8
Delhi           30882.80        5147.13             18                 6
Jaipur          25534.00        3647.71             17                 7
Kolkata         20169.00        2881.29             24                 7
Jammu           16059.75        2676.62             11                 6
Ahmedabad       14788.70        2464.78            

## 13. Sales Analysis by Product

In [13]:
product_analysis = df.groupby("Product").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Quantity_Sold=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values("Total_Sales", ascending=False)

print("Products ranked by Total Sales:")
print(product_analysis)

                         Total_Sales  Average_Sales  Quantity_Sold  Number_of_Orders
Product                                                                             
Coffee Maker                51610.25        7372.89             16                 7
Jacket                      47106.15        9421.23             20                 5
Cricket Bat                 41483.40        8296.68             18                 5
Running Shoes               41285.25        8257.05             15                 5
Wireless Headphones         33877.40        4839.63             23                 7
Dumbbell Set                30184.90        6036.98             16                 5
Mechanical Keyboard         26489.40        8829.80             11                 3
Electric Kettle             26382.40        5276.48             19                 5
Mixer Grinder               25341.55        6335.39              9                 4
Smart Watch                 15670.25        5223.42              

## 14. Sales Analysis by Payment Method

In [14]:
payment_analysis = df.groupby("Payment_Method").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Quantity_Sold=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values("Total_Sales", ascending=False)

print(payment_analysis)

                  Total_Sales  Average_Sales  Quantity_Sold  Number_of_Orders
Payment_Method                                                               
Credit Card         106540.25        4842.74             65                22
Debit Card           98064.90        4903.25             64                20
Net Banking          96371.25        5668.90             63                17
UPI                  96163.25        3698.59             70                26
Cash on Delivery     50123.30        3341.55             32                15


## 15. Sales Analysis by Region

In [15]:
region_analysis = df.groupby("Region").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Quantity_Sold=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values("Total_Sales", ascending=False)

print(region_analysis)

        Total_Sales  Average_Sales  Quantity_Sold  Number_of_Orders
Region                                                             
North     164989.45        4341.83            102                38
South     157351.45        5245.05             97                30
West       98155.05        4089.79             69                24
East       26767.00        3345.88             26                 8


## 16. Top-Performing Products

In [16]:
top_products = product_analysis.head(5)

print("Top 5 products by total sales:")
print(top_products)

                     Total_Sales  Average_Sales  Quantity_Sold  Number_of_Orders
Product                                                                         
Coffee Maker            51610.25        7372.89             16                 7
Jacket                  47106.15        9421.23             20                 5
Cricket Bat             41483.40        8296.68             18                 5
Running Shoes           41285.25        8257.05             15                 5
Wireless Headphones     33877.40        4839.63             23                 7


## 17. Highest-Selling Category and City

In [17]:
best_category = category_analysis["Total_Sales"].idxmax()
best_city = city_analysis["Total_Sales"].idxmax()
best_product = product_analysis["Total_Sales"].idxmax()
best_payment = payment_analysis["Total_Sales"].idxmax()
best_region = region_analysis["Total_Sales"].idxmax()

print("Highest-selling category:", best_category)
print("Highest-selling city:", best_city)
print("Highest-selling product:", best_product)
print("Payment method with highest total sales:", best_payment)
print("Highest-selling region:", best_region)

Highest-selling category: Sports
Highest-selling city: Hyderabad
Highest-selling product: Coffee Maker
Payment method with highest total sales: Credit Card
Highest-selling region: North


## 18. Meaningful Observations

1. The dataset contains 100 orders and generates total sales of ₹447,262.95.
2. The average sales value per order is ₹4,472.63, while the highest single order is ₹13,995.00.
3. Sports is the top-performing category by total sales, generating ₹136,026.45.
4. Coffee Maker is the highest-selling product by total sales, generating ₹51,610.25.
5. Hyderabad records the highest total sales among the cities, with ₹59,076.45.
6. Credit Card contributes the highest total sales among payment methods, with ₹106,540.25.
7. The dataset records a total quantity sold of 294 units across 100 orders.
8. There are no missing values in any of the 13 columns, so the dataset is complete for the analyses performed here.

## 19. Conclusion

The e-commerce dataset was successfully analyzed using Pandas. Data selection and filtering were used to identify relevant orders, while sorting helped find the highest- and lowest-value transactions. `groupby()` and aggregation functions provided comparisons across categories, cities, products, regions, and payment methods.

The analysis shows which products and categories contribute most to revenue and highlights differences in sales performance across locations and payment methods. These findings can be used to support product promotion, inventory planning, and sales strategy decisions.
